### PR TITLE
GEOT-5537: pass NOT instead of AND when creating NOT filters

### PR DESCRIPTION
**Pull request by mtoonen**
Unit test and fix for GEOT-5537 added: pass NOT filtertype to createHighLevelLogicFilter() method instead of AND.

### CODE REVIEW

File: modules/library/xml/src/test/java/org/geotools/xml/filter/FilterCompliancePreProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2002-20017, Open Source Geospatial Foundation (OSGeo)
```
Comment (mprins) [inline modules/library/xml/src/test/java/org/geotools/xml/filter/FilterCompliancePreProcessorTest.java]: That's a long time in the future :-)

**Comment by mprins** [comment]
Would be good to get this into Monday's? 16.2  as well

File: modules/library/xml/src/test/java/org/geotools/xml/filter/FilterCompliancePreProcessorTest.java (hunk 1 above)
Comment (bradh) [inline modules/library/xml/src/test/java/org/geotools/xml/filter/FilterCompliancePreProcessorTest.java]: Also I don't think OSGeo existed that far back (incorporated in 2006), so perhaps just this year...

**Comment by mtoonen** [comment]
License header adjusted.

**Pull request merged**